### ISSUE TITLE
custom criteria keys should be listed in `keys()`

### ISSUE THREAD

**Issue by kthyng** [xarray-contrib/cf-xarray#531]
I think it would be useful if for a defined custom criteria vocabulary (which can have many items defined), those that have found matches for a given Dataset have keys listed in [ds.cf.keys()](https://cf-xarray.readthedocs.io/en/latest/generated/xarray.Dataset.cf.keys.html#xarray.Dataset.cf.keys) so that it can be checked to see if a specific custom key is present in a Dataset. 

What do others think?

**Comment by dcherian** [comment]
Yes, similarly https://github.com/xarray-contrib/cf-xarray/issues/375
